.github/workflows/test-and-lint.yml

```diff
@@ -0,0 +1,115 @@
+# .github/workflows/test-and-lint.yml
+
+name: Test and Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # This job runs all the linting and formatting checks in parallel on a single OS.
```

Comment:
I don't really see anything that makes the linting and formatting process to be run in parallel. Am I missing something? Or is this referring to the `test` job?